node.go

```diff
@@ -487,7 +487,8 @@ func fieldByName(node Node, name string) Field {
 	return nil
 }
 
-func nodesAreEqual(node1, node2 Node) bool {
+// NodesAreEqual compares two parquet nodes and returns true if they are equal.
+func NodesAreEqual(node1, node2 Node) bool {
```

Comment:
Since we already have `Equal(Value, Value) bool`, how about calling this function `EqualNodes` to keep it a bit more consistent?

Reply:
Agreed. The `Equal(Value, Value) bool` GoDoc also helped me improve the GoDoc for `EqualNodes`.